rmw_zenoh_cpp/src/rmw_init_options.cpp

```diff
@@ -40,14 +41,14 @@ rmw_init_options_init(rmw_init_options_t * init_options, rcutils_allocator_t all
     return RMW_RET_INVALID_ARGUMENT;
   }
 
+  memset(init_options, 0, sizeof(rmw_init_options_t));
```

Comment:
**Suggestion:** Replace `memset` with a safer alternative like `std::fill_n` to avoid potential undefined behavior with non-trivial types. [best practice, importance: 8]
```suggestion
  std::fill_n(reinterpret_cast<char*>(init_options), sizeof(rmw_init_options_t), 0);
```